tests/test_kb_dataset.py

```diff
@@ -0,0 +1,310 @@
+                        ids_seen.add(v)
+        elif isinstance(collection, list):
+            for elem in collection:
+                self._assert_ids_globally_unique(elem, ids_seen)
+
+    def _get_referenced_ids(self, example):
+        referenced_ids = []
+
+        for event in example["events"]:
+            for argument in event["arguments"]:
+                referenced_ids.append((argument["ref_id"], "event_arg"))
+
+        for coreference in example["coreferences"]:
+            for entity_id in coreference["entity_ids"]:
+                referenced_ids.append((entity_id, "entity"))
+
+        for relation in example["relations"]:
+            referenced_ids.append((relation["arg1_id"], "entity"))
+            referenced_ids.append((relation["arg2_id"], "entity"))
+
+        return referenced_ids
+
+    def _get_existing_referable_ids(self, example):
+        existing_ids = []
+
+        for entity in example["entities"]:
+            existing_ids.append((entity["id"], "entity"))
+
+        for event in example["events"]:
+            existing_ids.append((event["id"], "event"))
+
+        return existing_ids
+
+    def test_passages_offsets(self):
+        """
+        Verify that the passages offsets are correct,
+        i.e.: passage text == text extracted via the passage offsets
+        """
+
+        for split in self.dataset_bigbio:
+
+            if "passages" in self.dataset_bigbio[split].features:
+
+                for example in self.dataset_bigbio[split]:
+
+                    example_text = _get_example_text(example)
+
+                    for entity in example["passages"]:
+                        for idx, (start, end) in enumerate(entity["offsets"]):
+                            self.assertEqual(
+                                example_text[start:end], entity["text"][idx]
+                            )
+
+    def test_entities_offsets(self):
+        """
+        Verify that the entities offsets are correct,
+        i.e.: entity text == text extracted via the entity offsets
+        """
+
+        errors = []
+
+        for split in self.dataset_bigbio:
+
+            if "entities" in self.dataset_bigbio[split].features:
+
+                for example in self.dataset_bigbio[split]:
+
+                    example_id = example["id"]
+                    example_text = _get_example_text(example)
+
+                    for entity in example["entities"]:
+
+                        for msg in _check_offsets(
+                            example_text=example_text,
+                            offsets=entity["offsets"],
+                            texts=entity["text"],
+                        ):
+
+                            entity_id = entity["id"]
+                            errors.append(
+                                f"Example:{example_id} - entity:{entity_id} " + msg
+                            )
+
+        if len(errors) > 0:
+            self.fail(msg="\n".join(errors) + OFFSET_ERROR_MSG)
```

Comment:
I really like the idea of collecting some examples to show to the user .. very cool!

Reply:
Thanks! People are not asked to fix the offsets but this turns useful in case they wanted to.